Rubberduck.CodeAnalysis/Inspections/Concrete/UnreachableCaseInspection/ParseTreeExpressionEvaluator.cs

```diff
@@ -61,32 +66,79 @@ public class ParseTreeExpressionEvaluator : IParseTreeExpressionEvaluator
             Tokens.Integer,
             Tokens.Byte,
             Tokens.Boolean,
+            Tokens.String
         };
 
-        public ParseTreeExpressionEvaluator(IParseTreeValueFactory valueFactory)
+        public ParseTreeExpressionEvaluator(IParseTreeValueFactory valueFactory, bool isOptionCompareBinary = true)
         {
             _valueFactory = valueFactory;
+            _isOptionCompareBinary = isOptionCompareBinary;
```

Comment:
So, if we review the documentation for the particular overload of `string.Compare` ([here](https://msdn.microsoft.com/en-us/library/hyxc48dt(v=vs.110).aspx)), we see a problem - it will consider `Animal` and `ani-mal` equivalent, which is not how VBA works. 

I think we should be using [this overload](https://msdn.microsoft.com/en-us/library/cc190529(v=vs.110).aspx) instead. There, you can pass in both the `CultureInfo` and `CompareOption` to better correspond to the locale that is used by the VBA project. For this PR, you only need to support `CompareOptions.Ordinal` and `CompareOptions.OrdinalIgnoreCase`, while make it easy to update the code to pass in a different `CultureInfo`/`CompareOption` when we do support locale-specific comparison in future. 

Does that help?

Reply:
@bclothier thanks.  Yes that helped a lot.  I've updated the string compare portions to:

        `private bool AreEqual(string lhs, string rhs)
        {
            var compareOptions = _isOptionCompareBinary ? 
                StringComparison.CurrentCulture | StringComparison.Ordinal 
                : StringComparison.CurrentCulture | StringComparison.OrdinalIgnoreCase;
            return String.Equals(lhs, rhs, compareOptions);
        }

        private bool IsLessThan(string lhs, string rhs)
        {
            var compareOptions = _isOptionCompareBinary ? CompareOptions.None : CompareOptions.IgnoreCase;
            return String.Compare(lhs, rhs, CultureInfo.CurrentCulture, compareOptions) < 0;
        }

        private bool IsGreaterThan(string lhs, string rhs)
        {
            var compareOptions = _isOptionCompareBinary ? CompareOptions.None : CompareOptions.IgnoreCase;
            return String.Compare(lhs, rhs, CultureInfo.CurrentCulture, compareOptions) > 0;
        }`

Hopefully I've interpreted the documentation correctly.